pymc3/tests/test_step.py

```diff
@@ -238,3 +241,129 @@ def test_binomial(self):
             steps = assign_step_methods(model, [])
         self.assertIsInstance(steps, Metropolis)
+
+
+class TestSampleEstimates(SeededTest):
+    def test_parameter_estimate(self):
+        alpha_true, sigma_true = 1, 0.5
+        beta_true = np.array([1, 2.5])
+
+        size = 100
+
+        X1 = np.random.randn(size)
+        X2 = np.random.randn(size) * 0.2
+        Y = alpha_true + beta_true[0] * X1 + beta_true[1] * X2 + np.random.randn(size) * sigma_true
+
+        with Model() as model:
+            alpha = Normal('alpha', mu=0, sd=10)
+            beta = Normal('beta', mu=0, sd=10, shape=2)
+            sigma = Uniform('sigma', lower=0.0, upper=1.0)
+            mu = alpha + beta[0] * X1 + beta[1] * X2
+            Y_obs = Normal('Y_obs', mu=mu, sd=sigma, observed=Y)
+
+            for step_method in (NUTS(), Metropolis(),
+                                [Slice([alpha, sigma]), Metropolis([beta])]):
+                trace = sample(100000, step=step_method, progressbar=False)
+
+                np.testing.assert_array_almost_equal(np.median(trace.beta, 0), beta_true, decimal=1)
+                np.testing.assert_array_almost_equal(np.median(trace.alpha), alpha_true, decimal=1)
+                np.testing.assert_array_almost_equal(np.median(trace.sigma), sigma_true, decimal=1)
+
+    def test_posterior_estimate_metropolis(self):
+        alpha_true, sigma_true = 1, 0.5
+        beta_true = np.array([1, 2.5])
+
+        size = 100
+
+        X1 = np.random.randn(size)
+        X2 = np.random.randn(size) * 0.2
+        Y = alpha_true + beta_true[0] * X1 + beta_true[1] * X2 + np.random.randn(size) * sigma_true
+
+        with Model() as model:
+            alpha = Normal('alpha', mu=0, sd=10)
+            beta = Normal('beta', mu=0, sd=10, shape=2)
+            sigma = Uniform('sigma', lower=0.0, upper=1.0)
+            mu = alpha + beta[0] * X1 + beta[1] * X2
+            Y_obs = Normal('Y_obs', mu=mu, sd=sigma, observed=Y)
+
+            trace = sample(100000, step=Metropolis(), progressbar=False)
+
+            # Using the Normal test from SciPy `scipy.stats.normaltest`
+            # See `https://github.com/scipy/scipy/blob/v0.18.1/scipy/stats/stats.py#L1352`
+            # `normaltest` returns a 2-tuple of the chi-squared statistic, and the associated
+            # p-value. Given the null hypothesis that `trace[-1000:]` came from the a normal
+            # distribution, the p-value represents the probability that a chi-squared statistic
+            # that large (or larger) would be seen.
+
+            # If the p-value is very small, it means it's unlikely that the data came from a
+            # normal distribution.
+            # We define very small as 0.05, for the sake of this argument.
+            # We test this by evaluating if the p-value is above 0.05.
+            _, test_normal = stats.normaltest(trace[-3000:].alpha)
+            np.testing.assert_array_less(0.05, test_normal, verbose=True)
```

Comment:
So this tests that `.05 < p_normal` which is what we want, so we get a nice, normal posterior. I would add this test for alpha, beta_0 and beta_1.